Source/WarpX.H

```diff
@@ -819,6 +819,12 @@ protected:
     //! data.  Called by AmrCore::regrid.
     virtual void RemakeLevel (int lev, amrex::Real time, const amrex::BoxArray& ba,
                               const amrex::DistributionMapping& dm) final;
+    virtual void RemakeMultiFab (std::unique_ptr<amrex::MultiFab>& mf,
+                                 const amrex::DistributionMapping& dm,
+                                 const bool redistribute = true) final;
+    virtual void RemakeiMultiFab (std::unique_ptr<amrex::iMultiFab>& mf,
+                                 const amrex::DistributionMapping& dm,
+                                 const bool redistribute = true) final;
```

Comment:
A couple comments on these
* The only member variable I can see that they access is `ng`, so if that were passed as an argument, they could be  free functions (possibly `static`) in `WarpXRegrid.cpp`.
* There's no difference between the bodies of these two overloads. A single `template <typename MultiFabType> void RemakeMultiFab(std::unique_ptr<MultiFabType>& mf, ...)` would cover both.
* Just a general C++ note - it's rather silly to declare a member function `virtual ... final` all in one go. If it's the only possible implementation, it just shouldn't be `virtual`

Reply:
Thanks for the comments! On the first point: there are also calls to `mf->boxArray()` and `mf->nComp()` besides the `ng = mf->nGrowVect()` line, so I think it is best to keep the arguments to `RemakeMultiFab` as is. For the other comments I just added a commit with the suggested changes.

Reply:
Oops, sorry, I'm silly. I missed that `ng` is a local variable, not a member of `WarpX` at all.

If you move the template definition up above the call sites in the .cpp file, then it doesn't need to appear in the header at all. And of course, need not be a member of WarpX

Reply:
Good point, thanks!